server/lib/pbench/indexer.py

```diff
@@ -3203,14 +3203,17 @@ def mk_sosreports(tb, extracted_root, logger):
         except Exception as e:
             logger.warning("Failed to fetch .md5 of sosreport {}: {}", sos, e)
             continue
-        host_md = hostnames_if_ip_from_sosreport(os.path.join(extracted_root, sos))
+        ret_val = hostnames_if_ip_from_sosreport(os.path.join(extracted_root, sos))
         # get hostname (short and FQDN) from sosreport
         d = _dict_const()
         d['name'] = sos
         d['md5'] = md5_val
-        d.update(host_md)
-        sosreportlist.append(d)
-
+        if ret_val[0] == 0:
+            d.update(ret_val[1])
+            sosreportlist.append(d)
+        else:
+            d['sos-report-error'] = ret_val[1]
```

Comment:
I'd have spelled it 'sosreport-error', but it's OK either way.  

You  can move the `sosreportlist.append(d) ` after the if-statement: both branches do the same thing.

Reply:
sure